Update language to be more mf-clear (#3883)
fishtown-analytics/docs.getdbt.com: website/docs/guides/best-practices/how-we-build-our-metrics/semantic-layer-1-intro.md

## Learning goals

> - ❓ Understand the **purpose and capabilities** of the **dbt Semantic Layer**, particularly MetricFlow as the engine that powers it.

Reviewer: ```suggestion
- ❓ Understand the **purpose and capabilities** of the **dbt Semantic Layer**, with a focus on MetricFlow as its driving engine.
```

Reviewer: hey @gwenwindflower making this sugg -

- 🧱 Familiarity with the core components of MetricFlow — **semantic models and metrics** — and how they work together.
- 🛠️ Hands-on **experience building** semantic models and metrics in dbt Cloud.
- 🔁 Know how to **refactor** models for MetricFlow.

> - 🏅 Aware of new **best practices** to take maximum advantage of the Semantic Layer.

Reviewer: this is prob a PMM question but I think the SL is referenced a few diff ways so do we want it to be dbt Semantic Layer or dbt Cloud Semantic Layer or Semantic Layer?
